### ISSUE TITLE
Disable editor horizontal scrollbar

### ISSUE THREAD

**Issue by SamVerschueren** [machinelabs/machinelabs#785]
#### Bug, feature request, or proposal:

Proposal

#### What is the expected behavior?

Disable the horizontal scroll bar of the editor when the code exceeds the width of the screen.

#### What is the current behavior?

Currently, when the code exceeds the width of the screen, the editor shows a horizontal scroll bar.

#### What are the steps to reproduce?

Just open a lab and start typing untill you exceed the width of the screen.

#### What is the use-case or motivation for changing an existing behavior?

Scrolling horizontally is very hard in a browser on mac because the browser thinks you are navigating back or forward. I didn't wrote a lot of code myself in the editor, but experienced this while I was solving #776. Maybe @cburgdorf and @d3lm have a better feeling with this.

I also noticed that StackBlitz disabled the horizontal scrolling and just starts on a new line. I'd rather not do this because it makes the code ugly when auto-wrapping lines.

#### Which version of MachineLabs (or the package) is affected ?

Latest

-----------------
I just want to gather feedback regarding this. I don't really have a strong opinion.

#### Cons
Code looks better, also on smaller screens as it is not auto-wrapped. Enabling auto-wrapping would make viewing code on mobile a mess.

#### Pros
Scrolling horizontally feels like a pain currently and I'm not sure if we can improve this as it's default on mac OS to navigate backwards and forwards in the browser history.

**Comment by PascalPrecht** [comment]
How do @cburgdorf and @d3lm feel about this?

**Comment by cburgdorf** [comment]
>I'd probably even vote for auto wrap as well (mainly cuz I'm used to that in my local editor setup)

When we turn off horizontal scrolling. What would be the alternative to auto wrap? Doesn't that implicitly mean we have to go with auto wrap to not cut of text?

I'm also in favor of disabling horizontal scrolling.

**Comment by SamVerschueren** [comment]
Yes, if we disable horizontal scrolling, it means auto-wrapping the content.

**Comment by cburgdorf** [comment]
>Yes, if we disable horizontal scrolling, it means auto-wrapping the content.

Alright, I think I'd be in favor of that even though it will screw code blocks on mobile. On that note, I think the main focus on mobile is stuff like checking execution progress / output etc and not code.

**Comment by PascalPrecht** [comment]
Maybe we can have both? Keep scrolling on mobile and disable it on tablet and up? We can use BreakpointObserver and a monaco API to control that.

**Comment by PascalPrecht** [comment]
I'd vote for keeping the scrolling on mobile because:

1. The motivation to change this behaviour originates from a desktop viewport/experience
2. Users are unlikely to "code" on their mobile phones
3. Reading code on mobile actually works very very well, and I'd like to keep it that way.

@SamVerschueren do have an API in ngx-monaco to turn this on and off?

**Comment by SamVerschueren** [comment]
@PascalPrecht the `MonacoEditorService` already exposes `updateOptions`. Using the `wordWrap` option should do the trick https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.ieditoroptions.html#wordwrap.

I agree that we should keep the horizontal scrollbar on mobile. Looking into this right now.
